cobigen-angular-client-generation/index.asciidoc

```diff
@@ -0,0 +1,95 @@
+= Generate Angular Client with devonfw
+== angular client generation with cobigen
+====
+With the CobiGen code generator you can easily generate components for your angular application.
+The following tutorial provides an example for using the CobiGen cli.
+You can find more information on https://github.com/devonfw/cobigen/wiki/howto_angular-client-generation
+## Prerequisites
+* devonfw Ide
+* Install yarn globally
+## Learning goals.
+Here in this tutorial we will learn the following:
+* Create an angular project
+* Cobigen to generate the angular source code out of the java files
+* Change the user interface in the angular project to show the new components
+* Show the newly created components
+====
+
+[step]
+--
+restoreDevonfwIde(["java", "mvn", "vscode", "node", "ng", "npm"])
+--
+
+For this tutorial an installation of the devonfw IDE is required, which was already set up for you. The next step is to install the CobiGen CLI.
+[step]
+--
+installCobiGen()
+--
+
+CobiGen uses the backend classes for generating your code used in the angular application. So in the next steps you will create a java project, create a simple java class and run the backend server.
+[step]
+--
+createDevon4jProject("cobigenjavaexample")
+--
+
+====
+[step]
+== Build and Run Java Server
+--
+buildJava("cobigenjavaexample")
+runServerJava("cobigenjavaexample/server",{ "startupTime": 180, "port": 8080, "path": "cobigenjavaexample" })
+--
+Now the Java Server should be running.
+====
+
+CobiGen is integrated via plugin in the VS Code IDE. We will use it to generate code from one single java class based on existing templates.
+[step]
+--
+createDevon4ngProject("cobigenangularexample", "")
```

Comment:
You can pass an array of arguments as third parameter to createDevon4ngProject to give some additional options. For example to set the style or the routing. Otherwise the user has to set this manually in katacoda.
createDevon4ngProject("cobigenangularexample", "", ["--style=scss", "--routing=true", "--strict=false"])

Reply:
> You can pass an array of arguments as third parameter to createDevon4ngProject to give some additional options. For example to set the style or the routing. Otherwise the user has to set this manually in katacoda.
> createDevon4ngProject("cobigenangularexample", "", ["--style=scss", "--routing=true", "--strict=false"])

added